### PR TITLE
✨Support Twitter timelines in amp-twitter.

### LINKED ISSUE #300: Add user timeline support to AMP-TWITTER
On implementation: 
1- Currently amp-twitter requires data-tweetid, we will introduce data-timelineid and make ONE of them required (changes required to validator-amp-twieer.protoascii)
2- amp-twitter.js most likely does not need to change as it just passes data to twitter.js in 3p folder
3- Inside twitter.js we would change export function twitter to check whether data.tweetid or data.timelineid exist and depending on which one is there, it would either do what it does now or fork and uses twttr.widgets.createTimeline instead of twttr.widgets.createTweet.
4- update documentation and tests, etc..

On contributing process: 
- [ ] If you aren't too familiar with Git/GitHub, see the [Getting Started End-to-End Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md) for [an intro to Git & GitHub,](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#intro-to-git-and-github) and [how to get a copy of the code](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#get-a-copy-of-the-amphtml-code).  You can also refer to the [Quick Start Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-quick.md) for the necessary setup steps with less explanation than the End-to-End guide.
- [ ] Follow the instructions for [building AMP](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server).
- [ ] [Create a Git branch](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#create-a-git-branch) for making your changes.
- [ ] Implement the solution.
- [ ] [Commit your changes](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#edit-files-and-commit-them) frequently.
- [ ] [Push your changes to GitHub](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#push-your-changes-to-your-github-fork).
- [ ] [Create a Pull Request](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#send-a-pull-request-ie-request-a-code-review).  Mention `closes Issue #300` in the description.
- [ ] Assign @aghassemi as the reviewer
- [ ] [Respond to your reviewer's comments](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#respond-to-pull-request-comments) (if any).

### PR DESCRIPTION
**Pull request by felixarntz**
* Adds a `data-timeline-source-type` attribute as mandatory alternative to `data-tweetid` and `data-momentid` to `<amp-twitter>`.
* Passes any additional `data-timeline-*` attributes to the `twttr.widgets.createTimeline` call after stripping the prefix, to specify further details.
* Updates validation to enforce that either `data-tweetid`, `data-momentid` or `data-timeline-source-type` is specified.
* Updates validation to support the extra parameters that can be used for a Twitter timeline.

Fixes #300 
/cc @sparhami

### CODE REVIEW

**Comment by Gregable** [comment]
Validation changes look good.

File: extensions/amp-twitter/amp-twitter.md [hunk 1]
```diff
@@ -93,16 +93,22 @@ Visit the [Placeholders & fallbacks](https://www.ampproject.org/docs/guides/resp
...
 ## Attributes
 
-##### data-tweetid / data-momentid (required)
+##### data-tweetid / data-momentid / data-timeline-source-type (required)
 
-The ID of the Tweet or Moment.
+The ID of the Tweet or Moment, or the source type if a Timeline should be displayed.
 In a URL like https://twitter.com/joemccann/status/640300967154597888,  `640300967154597888` is the tweet id.
 In a URL like https://twitter.com/i/moments/1009149991452135424, `1009149991452135424` is the moment id.
+Valid timeline source types include `profile`, `likes`, `list`, `collection`, `url`, or `widget`.
```
Comment (sparhami) [inline extensions/amp-twitter/amp-twitter.md]: or -> and

**Comment by felixarntz** [comment]
I've updated the PR with the suggested changes.

**Comment by sparhami** [comment]
Thanks for taking a look at this Felix! I went ahead and updated the validator tests and merged as everything else looked good.

**Comment by bjimdev** [comment]
I think that this component have a bug managing the height when use the timeline option u can try to put this peace of code into the amp playground and try for yourself.

```
<amp-twitter width="486" height="300"
      layout="fixed"
      data-timeline-source-type="profile"
      data-timeline-screen-name="amphtml"
      data-tweet-limit="5">
  </amp-twitter>
```

**Comment by aghassemi** [comment]
@bjimdev This component, similar to `amp-twitter`, resizes the height to always fit the content.

**Pull request merged**